rcl_yaml_param_parser/src/parse.c

```diff
@@ -89,7 +90,8 @@ void * get_value(
 
   /// Check if it is bool
   if (style != YAML_SINGLE_QUOTED_SCALAR_STYLE &&
-    style != YAML_DOUBLE_QUOTED_SCALAR_STYLE)
+    style != YAML_DOUBLE_QUOTED_SCALAR_STYLE &&
+    (tag == NULL || strcmp(YAML_STR_TAG, (char *)tag) != 0))
```

Comment:
It seems the `strcmp` will be called multiple times if using `!!str` in the yaml.

Reply:
Nice catch, that we can improve!